Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -479,17 +493,17 @@ inline bool ShouldDraw() {
         return SUCCEEDED(_firstErrorHr) && CurrentGState().ShouldDraw();
     }
 
-    inline void PushBeginDraw() {
-        if ((_beginEndDrawDepth)++ == 0) {
+    inline void PushBeginDraw(bool forceBeginDraw = false) {
+        if (forceBeginDraw || ((_beginEndDrawDepth)++ == 0)) {
             if (SUCCEEDED(_firstErrorHr)) {
                 deviceContext->BeginDraw();
             }
         }
     }
 
-    inline HRESULT PopEndDraw() {
+    inline HRESULT PopEndDraw(bool forceEndDraw = false) {
```

Comment:
The whole point was to make it more usable, we don't want to start making private functions for every single case.
This seems like too much dancing around to do the thing we are going to do any way.

---
In reply to: [106077266](https://github.com/Microsoft/WinObjC/pull/2208#discussion_r106077266) [](ancestors = 106077266)